components/safety_board/src/SYS_Vehicle.c

```diff
@@ -0,0 +1,255 @@
+/**
+ * @file SYS_Vehicle.c
+ * @brief  Source code for the Vehicle's Safety State Machine
+ * @author Joshua Lafleur (josh.lafleur@outlook.com)
+ * @version 0.1
+ * @date 2023-01-16
+ */
+
+/******************************************************************************
+ *                             I N C L U D E S
+ ******************************************************************************/
+
+#include "SYS_Vehicle.h"
+
+#include "SystemConfig.h"
+#include <stdint.h>
+
+#include "HW_can.h"
+#include "HW_gpio.h"
+
+
+/******************************************************************************
+ *                         P R I V A T E  V A R S
+ ******************************************************************************/
+
+static uint8_t          status_flags      = 0;
+static VEHICLE_STATES_E vehicle_state     = VEHICLE_INIT;
+static int16_t          ts_voltage        = 0;    // Stored at 0.1 V increments
+static int16_t          batt_voltage      = 0;    // Stored at 0.1 V increments
+static uint32_t         last_ts_message   = 0;
+static uint32_t         last_batt_message = 0;
+static uint8_t          imd_duty_cycle = 0;
+static uint8_t          imd_freq = 0;
+
+
+/******************************************************************************
+ *          P R I V A T E  F U N C T I O N  P R O T O T Y P E S
+ ******************************************************************************/
+
+bool SYS_Validate(void);
+
+/******************************************************************************
+ *                       P U B L I C  F U N C T I O N S
+ ******************************************************************************/
+
+void SYS_SAFETY_Init(void)
+{
+    SYS_CONTACTORS_OpenAll();
+
+    if (HAL_GPIO_ReadPin(OK_HS_Port, OK_HS_Pin) == GPIO_PIN_SET)
+    {
+        SYS_SAFETY_SetStatus(IMD_STATUS, ON);
+    }
+    else
+    {
+        SYS_SAFETY_SetStatus(IMD_STATUS, OFF);
+    }
+
+    if (HAL_GPIO_ReadPin(TSMS_CHG_Port, TSMS_CHG_Pin) == GPIO_PIN_SET)
+    {
+        SYS_SAFETY_SetStatus(TSMS_STATUS, ON);
+    }
+    else
+    {
+        SYS_SAFETY_SetStatus(TSMS_STATUS, OFF);
+    }
+}
+
+void SYS_CONTACTORS_OpenAll(void)
+{
+    HAL_GPIO_WritePin(AIR_Port, AIR_Pin, GPIO_PIN_RESET);
+    HAL_GPIO_WritePin(PCHG_Port, PCHG_Pin, GPIO_PIN_RESET);
+}
+
+bool SYS_SAFETY_GetStatus(STATUS_INDEX_E status)
+{
+    return (status_flags & (0x01 << status));
```

Comment:
why not use the Flags module I build in the steering wheel code? We could move that out to the libs dir